### PR TITLE
add some public link tests

### PR DESCRIPTION
**Pull request by wkloucek**

### CODE REVIEW

**Comment by micbar** [comment]
Good progress. I made the linter pass.

**Comment by micbar** [comment]
@refs This is our current approach. If you have ideas to make it easier, please let us know

File: featurecontext/init.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	providerv1beta1 "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
+	"github.com/cs3org/reva/pkg/rgrpc/todo/pool"
+)
+
+func (f *FeatureContext) Init(endpoint string, httpInsecure bool) {
+
+	client, err := pool.GetGatewayServiceClient(endpoint)
+	if err != nil {
+		// how to handle this?
+		fmt.Println("couldnt open a connection")
```
Comment (C0rby) [inline featurecontext/init.go]: This probably shouldn't continue without a connection.

Comment (micbar) [reply featurecontext/init.go]: added logging a fatal error to exit the program

File: featurecontext/auth.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+import (
+	"context"
+	"fmt"
+
+	ctxpkg "github.com/cs3org/reva/pkg/ctx"
+	"google.golang.org/grpc/metadata"
+)
+
+// getAuthContext uses the access token from the Feature Context
```
Comment (refs) [inline featurecontext/auth.go]:
```suggestion
// GetAuthContext uses the access token from the Feature Context
```

File: go.mod [hunk 3]
```diff
@@ -1,32 +1,39 @@
-module cs3api-validator
+module github.com/owncloud/cs3api-validator
```
Comment (refs) [inline go.mod:1]: 😄

File: go.mod [hunk 4]
```diff
@@ -1,32 +1,39 @@
...
 	github.com/stretchr/testify v1.7.0
 	google.golang.org/grpc v1.42.0
 )
 
 require (
+	github.com/Azure/go-ntlmssp v0.0.0-20200615164410-66371956d46c // indirect
 	github.com/cucumber/gherkin-go/v19 v19.0.3 // indirect
 	github.com/cucumber/messages-go/v16 v16.0.1 // indirect
 	github.com/davecgh/go-spew v1.1.1 // indirect
+	github.com/go-asn1-ber/asn1-ber v1.5.1 // indirect
```
Comment (refs) [inline go.mod:29]: heck, do we really need this? this is LDAP low level stuff 🤔

File: scenario/featurecontext.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+type featureContext struct {
+	*featurecontext.FeatureContext
+
+	*login.LoginFeatureContext
+	*publicshare.PublicShareFeatureContext
+	*resources.ResourcesFeatureContext
+	*spaces.SpacesFeatureContext
+}
+
+type StepsRegistrar interface {
```
Comment (refs) [inline scenario/featurecontext.go]: `StepsRegistrar` is implemented in the FeatureContext(s) but not really used 🤔 if it is dead code, it clutters a bit.

**Review by refs — APPROVED** [review]

**Comment by micbar** [comment]
@butonic @kobergj I added a workaround for the listStoragespaces.

I found out it doesn't work without filters. We should check why.

File: scenario/featurecontext.go (hunk 5 above)
Comment (micbar) [reply scenario/featurecontext.go]: removed it

File: go.mod (hunk 4 above)
Comment (micbar) [reply go.mod:29]: indirect dependency of godog?

File: featurecontext/auth.go (hunk 2 above)
Comment (micbar) [reply featurecontext/auth.go]: good catch

**Review by micbar — APPROVED** [review]

**Pull request merged**